### DISCUSSION TITLE
~ and ` should move to the next character if selection is just 1 char wide

### DISCUSSION THREAD

**Discussion by vp2177** [helix-editor/helix#6769] [Ideas]
1. Given the text `computerid`, for example
2. Place the cursor on `i`, and press `~`
3. Result: `computerId`, with the cursor remaining on `I`
4. **Expected**: Being able to press `~` again to get `computerID`, that is, the cursor being moved to the right by `~` if the selection's size was just 1 character.

I think this would improve usability of this common use-case

**Comment by CptPotato** [comment]
I personally prefer the current behavior, but I'm curious what others think.

I also think the selection should either *always* or *never* be moved, instead of introducing a special case for one-character selections.

Edit: You could also get the behavior you describe using a custom key bind: `[ "switch_to_uppercase", "move_char_right" ]`.

**Comment by vp2177** [comment]
@CptPotato Consistency is good (_never_ moving the cursor), I think this case is an exception though but maybe I'm in the minority
